doc/fcn_2D_segm.txt

```diff
@@ -0,0 +1,205 @@
+transposed convolution layer with stride 32.
+
+2. **FCN-16** : Sums the 2x upsampled prediction from *conv7*
+(using a transposed convolution with stride 2) with *pool4* and then
+produces the segmentation map, by using a transposed convolution layer with stride 16
+on top of that.
+
+3. **FCN-8** : Sums the 2x upsampled *conv7* (with a stride 2 transposed convolution)
+with *pool4*, upsamples them with a stride 2 transposed convolution and sums them
+with *pool3*, and applies a transposed convolution layer with stride 8 on the resulting
+feature maps to obtain the segmentation map.
+
+
+.. figure:: images/fcn_schema.png
+    :align: center
+    :scale: 65%
+
+    **Figure 3** : FCN architecture (from FCN paper)
+
+As explained above, the upsampling paths of the FCN variants are different, since they
+use different skip connection layers and strides for the last convolution, yielding
+different segmentations, as shown in Figure 4. Combining layers that have different
+precision helps retrieving fine-grained spatial information, as well as coarse
+contextual information.
+
+.. figure:: images/fcn32_16_8.png
+    :align: center
+    :scale: 30%
+
+    **Figure 4** : FCN results (from FCN paper)
+
+Note that the FCN-8 architecture was used on the polyps dataset below,
+since it produces more precise segmentation maps.
+
+
+Metrics
+=======
+
+**Per pixel accuracy**
+
+This metric is self explanatory, since it outputs the class prediction accuracy
+per pixel.
+
+.. math::
+   :label: jaccard
+
+    acc(P, GT) = \frac{|\text{pixels correctly predicted}|}{|\text{total nb of pixels}|}
+
+
+**Jaccard (Intersection over Union)**
+
+This evaluation metric is often used for image segmentation, since it is more structured.
+The jaccard is a per class evaluation metric, which computes the number of pixels in
+the intersection between the
+predicted and ground truth segmentation maps for a given class, divided by the
+number of pixels in the union between those two segmentation maps,
+also for that given class.
+
+.. math::
+   :label: jaccard_equation
+
+    jacc(P(class), GT(class)) = \frac{|P(class)\cap GT(class)|}{|P(class)\cup GT(class)|}
+
+where `P` is the predicted segmentation map and `GT` is the ground
+truth segmentation map. `P(class)` is then the binary mask indicating if each
+pixel is predicted as *class* or not. In general, the closer to 1, the better.
+
+.. figure:: images/jaccard.png
+    :align: center
+    :scale: 40%
+
+    **Figure 5** : Jaccard visualisation (from this `website <http://www.pyimagesearch.com/2016/11/07/intersection-over-union-iou-for-object-detection/>`__)
+
+Code
+++++
+
+The FCN-8 implementation can be found in the following file:
+
+* `fcn8.py  <../code/fcn_2D_segm/fcn8.py>`_  : Defines the model.
+* `train_fcn8.py <../code/fcn_2D_segm/train_fcn8.py>`_ : Training loop.
+
+
+TODO : import model_helpers, dataset_loader, metrics
+TODO : remove /Tmp/romerosa path and make them relative path
```

Comment:
I think we will need to talk about the datasets and their loaders, I'm not sure I understand what's up with that.